src/foam/nanos/dig/drivers/DigFormatDriver.js

```diff
@@ -0,0 +1,267 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.dig.drivers',
+  name: 'DigFormatDriver',
+  abstract: true,
+  flags: ['java'],
+
+  javaImports: [
+    'foam.core.*',
+    'foam.dao.AbstractDAO',
+    'foam.dao.ArraySink',
+    'foam.dao.DAO',
+    'foam.lib.csv.CSVOutputter',
+    'foam.lib.json.OutputterMode',
+    'foam.nanos.boot.NSpec',
+    'foam.nanos.dig.*',
+    'foam.nanos.dig.exception.*',
+    'foam.nanos.http.*',
+    'foam.nanos.logger.Logger',
+    'foam.nanos.logger.PrefixLogger',
+    'foam.mlang.MLang',
+    'foam.mlang.predicate.Predicate',
+    'foam.util.SafetyUtil',
+    'java.io.PrintWriter',
+    'java.util.ArrayList',
+    'java.util.Arrays',
+    'java.util.List',
+    'javax.servlet.http.HttpServletResponse'
+  ],
+
+  constants: [
+    {
+      name: 'MAX_PAGE_SIZE',
+      type: 'Long',
+      value: 1000
+    }
+  ],
+
+  properties: [
+    {
+      name: 'format',
+      class: 'Enum',
+      of: 'foam.nanos.http.Format'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'parseFObjects',
+      type: 'List',
+      javaThrows: [
+        'java.lang.Exception'
+      ],
+      args: [ 
```

Comment:
You could use the new args: format:
  args: [ 'X x', 'DAO dao', 'String data' ] 
if you like.